src/main/java/io/vertx/ext/mongo/util/CircularByteBuffer.java

```diff
@@ -0,0 +1,92 @@
+package io.vertx.ext.mongo.util;
+
+import java.nio.ByteBuffer;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class CircularByteBuffer {
+
+  private byte[] array;
+  private int sizeMask;
+  private final AtomicInteger writeSequence = new AtomicInteger(0);
+  private final AtomicInteger readSequence = new AtomicInteger(0);
+
+  public CircularByteBuffer(int size) {
```

Comment:
Is there a 3rd party circular byte buffer in Guava or some other dependency that vertx already has?  Nothing wrong with this impl but prefer not to have data structures inside the driver which may have battle tested variants elsewhere.

Reply:
I'll give CircularFifoQueue from Apache Commons a try. If I'm not mistaken, Julien doesn't care much for dependencies on Guava.

Reply:
Changed to Apache CircularFifoQueue